Sources/BraintreeCore/Analytics/BTAnalyticsService.swift

```diff
@@ -17,110 +19,52 @@ class BTAnalyticsService: Equatable {
-    ///   - errorDescription: Optional. Full error description returned to merchant.
-    ///   - isVaultRequest: Optional. If the Venmo or PayPal request is being vaulted.
-    ///   - linkType: Optional. The type of link the SDK will be handling, currently deeplink or universal.
-    ///   - payPalContextID: Optional. PayPal Context ID associated with the checkout session.
-    ///   - startTime: Optional. The start time of the networking request.
-    func sendAnalyticsEvent(
-        _ eventName: String,
-        connectionStartTime: Int? = nil,
-        correlationID: String? = nil,
-        endpoint: String? = nil,
-        endTime: Int? = nil,
-        errorDescription: String? = nil,
-        isVaultRequest: Bool? = nil,
-        linkType: String? = nil,
-        payPalContextID: String? = nil,
-        requestStartTime: Int? = nil,
-        startTime: Int? = nil
-    ) {
+    /// - Parameter event: A single `FPTIBatchData.Event`
+    func sendAnalyticsEvent(_ event: FPTIBatchData.Event) {
         Task(priority: .background) {
-            await performEventRequest(
-                eventName,
-                connectionStartTime: connectionStartTime,
-                correlationID: correlationID,
-                endpoint: endpoint,
-                endTime: endTime,
-                errorDescription: errorDescription,
-                isVaultRequest: isVaultRequest,
-                linkType: linkType,
-                payPalContextID: payPalContextID,
-                requestStartTime: requestStartTime,
-                startTime: startTime
-            )
+            await performEventRequest(event)
         }
     }
     
     /// Exposed to be able to execute this function synchronously in unit tests
-    func performEventRequest(
-        _ eventName: String,
-        connectionStartTime: Int? = nil,
-        correlationID: String? = nil,
-        endpoint: String? = nil,
-        endTime: Int? = nil,
-        errorDescription: String? = nil,
-        isVaultRequest: Bool? = nil,
-        linkType: String? = nil,
-        payPalContextID: String? = nil,
-        requestStartTime: Int? = nil,
-        startTime: Int? = nil
-    ) async {
-        let timestampInMilliseconds = Date().utcTimestampMilliseconds
-        let event = FPTIBatchData.Event(
-            connectionStartTime: connectionStartTime,
-            correlationID: correlationID,
-            endpoint: endpoint,
-            endTime: endTime,
-            errorDescription: errorDescription,
-            eventName: eventName,
-            isVaultRequest: isVaultRequest,
-            linkType: linkType,
-            payPalContextID: payPalContextID,
-            requestStartTime: requestStartTime,
-            startTime: startTime,
-            timestamp: String(timestampInMilliseconds)
-        )
-
-        await BTAnalyticsService.events.append(event)
+    func performEventRequest(_ event: FPTIBatchData.Event) async {
```

Comment:
Take it our leave it, could rename this to this since it reads a bit odd now (did before too):
```suggestion
    func performEventRequest(with event: FPTIBatchData.Event) async {
```